modules/frontend/frontend.go

```diff
@@ -161,14 +161,16 @@ func newTraceByIDMiddleware(cfg Config, logger log.Logger) Middleware {
 					}
 					resp.Body = io.NopCloser(bytes.NewReader(traceBuffer))
 				}
+
+				if resp.Header != nil {
```

Comment:
The containing if statement also checks for http status OK.  Does the header need to be set for errors as well?  I'm honestly unsure.

Reply:
I actually think this is better. Our errors tend to be just text so we probably shouldn't be setting the `Content-Type` header.